src/main/java/org/openrewrite/java/migrate/javax/AddScopeToInjectedClass.java

```diff
@@ -0,0 +1,96 @@
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.javax;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.ScanningRecipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+public class AddScopeToInjectedClass extends ScanningRecipe<Set<JavaType.FullyQualified>> {
+    private static final String JAVAX_INJECT_INJECT = "javax.inject.Inject";
+    private static final String JAVAX_ENTERPRISE_CONTEXT_DEPENDENT = "javax.enterprise.context.Dependent";
+    private static final Collection<String> TYPES_PROMPTING_SCOPE_ADDITION = Arrays.asList(JAVAX_INJECT_INJECT);
+
+    @Override
+    public String getDisplayName() {
+        return "Add scope annotation to injected classes";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Finds member variables annotated with `@Inject' and applies `@Dependent` scope annotation to the variable's type.";
+    }
+
+    @Override
+    public Set<JavaType.FullyQualified> getInitialValue(ExecutionContext ctx) {
+        return new HashSet<>();
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(Set<JavaType.FullyQualified> injectedTypes) {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext executionContext) {
+                J.ClassDeclaration cd = super.visitClassDeclaration(classDecl, executionContext);
+                for (JavaType.Variable variable : cd.getType().getMembers()) {
+                    if (variableTypeRequiresScope(variable)) {
+                        injectedTypes.add((JavaType.FullyQualified) variable.getType());
+                    }
+                }
+                return cd;
+            }
+
+            private boolean variableTypeRequiresScope(@Nullable JavaType.Variable memberVariable) {
+                if (memberVariable == null) {
+                    return false;
+                }
+
+                for (JavaType.FullyQualified annotation : memberVariable.getAnnotations()) {
+                    if (TYPES_PROMPTING_SCOPE_ADDITION.contains(annotation.getFullyQualifiedName())) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        };
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(Set<JavaType.FullyQualified> injectedTypes) {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.CompilationUnit visitCompilationUnit(J.CompilationUnit compilationUnit, ExecutionContext executionContext) {
```

Comment:
I suggest you instead also here override `visitClassDeclaration()`. That way it should automatically also be applicable to Groovy and Kotlin sources.

Reply:
The AnnotateTypesVisitor referenced in this method overrides `visitClassDeclaration()` - won't that have the same effect?

Reply:
Not quite. If your recipe were now to process a Kotlin source, it wouldn't do anything, because Kotlin sources don't contain `J.CompilationUnit` objects in their LST.

But since we don't have any Kotlin (or Groovy) tests to ensure that the recipe works correctly with these languages, we might as well leave it as-is.